nix/configuration.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+{
+  virtualisation.vmVariant = {
+    virtualisation = {
+      memorySize = 2048; # Use 2048MiB memory.
+      cores = 3;
```

Comment:
You can make that lighter by setting [virtualisation.forwardPort](https://search.nixos.org/options?channel=23.11&show=virtualisation.forwardPorts&from=0&size=50&sort=relevance&type=packages&query=virtualisation.forwardPort) to make the api available from the host (and avoid having to bundle a whole DM and web browser in the VM